### PR TITLE
Fill in gaps in misc fns docs according to clojuredocs

### PR DESCRIPTION
**Pull request by ccfontes**

### CODE REVIEW

File: docs/docs/reference/shards/lisp/misc.md [hunk 1]
```diff
@@ -33,44 +33,92 @@ This is useful for cases where you need multiple shards (to transform your data)
...
 
-```clojure linenums="1"
+```
+(apply f args) (apply f x args) (apply f x y args) (apply f x y z args)(apply f a b c d & args)
+```
+
+Applies fn f to the argument list formed by prepending intervening arguments to args.
 
+```clojure linenums="1"
+(apply str ["cherry " "banana " "apple"])
```
Comment (guusw) [inline docs/docs/reference/shards/lisp/misc.md:43]: Can you give an example with multiple arguments as well?
I got `(apply str "cherry " "banana " "apple" [])` to work but `(apply str "cherry " "banana " "apple")` doesn't.
I'm still a bit confused what this does.

File: docs/docs/reference/shards/lisp/misc.md [hunk 2]
```diff
@@ -33,44 +33,92 @@ This is useful for cases where you need multiple shards (to transform your data)
...
 ```
 
 ## atom
 
-```clojure linenums="1"
+```
+(atom x)
+```
 
+  Creates and returns an Atom with an initial value of x and zero or more options (in any order).
```
Comment (guusw) [inline docs/docs/reference/shards/lisp/misc.md:53]: Maybe add a brief description of what an atom is

File: docs/docs/reference/shards/lisp/misc.md [hunk 3]
```diff
@@ -33,44 +33,92 @@ This is useful for cases where you need multiple shards (to transform your data)
...
 ```
 
 ## deref
 
-```clojure linenums="1"
+```
+(deref ref)
+```
+
+Returns the most-recently-committed value of ref. When applied to an atom, returns its current state.
```
Comment (guusw) [inline docs/docs/reference/shards/lisp/misc.md:65]: What does `most-recently-committed` mean here? can you link it to an explanation or add a short one below.

File: docs/docs/reference/shards/lisp/misc.md [hunk 4]
```diff
@@ -33,44 +33,92 @@ This is useful for cases where you need multiple shards (to transform your data)
...
 ```
 
 ## macro?
 
-```clojure linenums="1"
+```
+(macro? form)
+```
 
+Checks if form is a macro.
```
Comment (guusw) [inline docs/docs/reference/shards/lisp/misc.md:117]: What's a form?

**Review by guusw — COMMENTED** [review]
The additions are okay, I just think there could be some more explanation about some of the terminology (either explained inline or linked to somewhere else).

If you want you can merge this and address it in another PR.

File: docs/docs/reference/shards/lisp/misc.md (hunk 3 above)
Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md:65]: I'll remove such troublesome explanation.

File: docs/docs/reference/shards/lisp/misc.md (hunk 4 above)
Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md:117]: > A form is a list whose first element is a symbol that names an operator. If the first element names a function, then the form is a functional form.

It excludes data data.

Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md:117]: Although this doesn't make sense in Mal.

File: docs/docs/reference/shards/lisp/misc.md (hunk 1 above)
Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md:43]: How about now?

File: docs/docs/reference/shards/lisp/misc.md (hunk 2 above)
Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md:62]: Is explain good?

File: docs/docs/reference/shards/lisp/misc.md (hunk 4 above)
Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md]: Now I mention value. Ima wa dou?

File: docs/docs/reference/shards/lisp/misc.md (hunk 1 above)
Comment (guusw) [reply docs/docs/reference/shards/lisp/misc.md:43]: Yeah looks good

File: docs/docs/reference/shards/lisp/misc.md (hunk 2 above)
Comment (guusw) [reply docs/docs/reference/shards/lisp/misc.md:62]: A bit, I'm asuming you'd use these to store data, so it's like this:
```clj
(def atom (atom 1))
(reset! atom 2)
```
?

**Review by guusw — APPROVED** [review]

File: docs/docs/reference/shards/lisp/misc.md (hunk 4 above)
Comment (guusw) [reply docs/docs/reference/shards/lisp/misc.md]: I think mal also uses the terminology 'form', I'm just no sure what it's definition is

Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md]: It's just a lispy umbrella for a lot of possible stuff under `()`. Like saying "object".

File: docs/docs/reference/shards/lisp/misc.md (hunk 2 above)
Comment (ccfontes) [reply docs/docs/reference/shards/lisp/misc.md:62]: alright, I improved with example.

**Review by guusw — APPROVED** [review]

**Pull request merged**